packages/react-query/src/shared/hooks/createHooksInternal.tsx

```diff
@@ -260,8 +260,14 @@ export function createRootHooks<
     pathAndInput: [path: string, ...args: unknown[]],
     opts?: UseTRPCQueryOptions<unknown, unknown, unknown, unknown, TError>,
   ): UseTRPCQueryResult<unknown, TError> {
+    const context = useContext();
+    if (!context) {
+      throw new Error(
+        'Unable to retrieve application context. Did you forget to wrap your App inside `withTRPC` HoC?',
```

Comment:
Wouldn't this also throw in a non-Next.js React app where the provider is missing? Maybe the message can be more general about missing the necessary context/provider?

Reply:
Thats exactly what it will do, thus my question about better wording 😅

Reply:
could it be better to link to an faq question that has possible answers of why it’s happening(withtrpc missing, provider missing, etc)

Reply:
Feel free to improve it in a PR, I'm merging this for now!